service/src/worker.rs

```diff
@@ -69,32 +68,28 @@ where
 
 		let peers = self.peers()?;
 		debug!("Gossiping sidechain blocks to peers: {:?}", peers);
+		let blocks_json = vec![to_json_value(blocks)?];
 
 		for p in peers.iter() {
 			// Todo: once the two direct servers are merged, remove this.
 			let url = worker_url_into_async_rpc_url(&p.url)?;
 			trace!("Gossiping block to peer with address: {:?}", url);
+			// FIXME: Websocket connectionto a worker  should stay once etablished.
 			let client = WsClientBuilder::default().build(&url).await?;
-			let response: String = client
-				.request::<Vec<u8>>(
-					"sidechain_importBlock",
-					vec![to_json_value(blocks.clone())?].into(),
-				)
-				.await
-				.map(String::from_utf8)?
-				.map(|s| s.trim_end().to_string())?; // remove whitespace padding
-			debug!("sidechain_importBlock response: {:?}", response);
+			let blocks = blocks_json.clone();
+			thread::spawn(move || async move {
+				if let Err(e) =
+					client.request::<Vec<u8>>("sidechain_importBlock", blocks.into()).await
+				{
+					debug!("sidechain_importBlock failed: {:?}", e);
```

Comment:
only `debug!` here, not `error!` ?

Reply:
I was actually thinking about that too. But in case a worker is, due to whatever reason, not receiving anything, do we really want to spam error logs? Once we have one connection per worker we can then close the connection, but right now we'd be spamming.

But as it's a temporary solution only, I can also live with error spamming. Should I adapt?